test/hotspot/jtreg/runtime/ErrorHandling/TestDwarf.java

```diff
@@ -123,18 +123,15 @@ private static void runAndCheck(Flags flags, DwarfConstraint... constraints) thr
         crashOut = ProcessTools.executeProcess(ProcessTools.createTestJvm(flags.getFlags()));
         String crashOutputString = crashOut.getOutput();
         Asserts.assertNotEquals(crashOut.getExitValue(), 0, "Crash JVM should not exit gracefully");
-        Pattern pattern = Pattern.compile("hs_err_pid[0-9]*.log");
-        Matcher matcher = pattern.matcher(crashOutputString);
         System.out.println(crashOutputString);
-        if (matcher.find()) {
-            String hsErrFileName = matcher.group();
-            System.out.println("hs_err_file: " + hsErrFileName);
-            File hs_err_file = new File(hsErrFileName);
-            BufferedReader reader = new BufferedReader(new FileReader(hs_err_file));
+
+        File hs_err_file = HsErrFileUtils.openHsErrFileFromOutput(crashOut);
+        try (BufferedReader reader = new BufferedReader(new FileReader(hs_err_file))) {
```

Comment:
Not sure this is the correct twr idiom when constructing multiple resources ie should it be:
```
try (var fr = new FileReader(hs_err_file);
      var reader = new BufferredReader(fr);) {
...
}
```